app/views/steps/case/case_type/edit.html.erb

```diff
@@ -0,0 +1,34 @@
+<% title t('.page_title') %>
+<% step_header %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= govuk_error_summary(@form_object) %>
+
+    <%= step_form @form_object do |f| %>
+      <%= f.govuk_radio_buttons_fieldset(:case_type, legend: {tag: 'h1', size: 'xl'}) do %>
+        <% @form_object.choices.each do |choice| %>
+          <% if choice.value == :cc_appeal %>
+
+            <%= f.govuk_radio_button :case_type, choice.value do %>
+              <%= f.govuk_text_field :cc_appeal_maat_id, width: 'one-third' %>
+            <% end %>
+
+          <% elsif choice.value == :cc_appeal_fin_change %>
+
+            <%= f.govuk_radio_button :case_type, choice.value do %>
+              <%= f.govuk_text_field :cc_appeal_fin_change_maat_id, width: 'one-third' %>
+              <%= f.govuk_text_area :cc_appeal_fin_change_details %>
+            <% end %>
+
+          <% else %>
+
+            <%= f.govuk_radio_button :case_type, choice.value, link_errors: true %>
```

Comment:
Unfortunately this still sets the `link_errors: true` in more than one radio.

I think a way to do it might be to iterate the collection yielding the index, and then only adding the link_errors to the index 0, this way we make sure only the first radio in the collection has this.

```ruby
@form_object.choices.each.with_index do |choice, index|
[...]
link_errors: index.zero?
```

The other alternative is to declare all the individual radios, and to not use a loop. In terms of lines of code it would be similar, maybe more readable as there will not be if-else etc. But is a balance of convenience as if the case types value object adds or removes case types then the view will need to be adapted too. Depends if we think the case types are going to change frequently 🤷‍♂️

Reply:
I think i'll go with the index option 👍 see if that's okay. I like the idea of the loop as we _can_ iterate it if you know what I mean ha